### PR TITLE
Refactor structs

### PR DESCRIPTION
**Pull request by tteerawat**
- Make the decoding process dynamically based on the returned `Omise-Version`value.
- Add an option for users to set `decode` boolean flag.
- Revise the tests.
- All functions work the same way as usual.

### CODE REVIEW

File: lib/omise/http_client.ex [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+      def put(endpoint, body_params \\ [], options \\ []) do
+        Omise.HTTPClient.request(:put, @key_type, endpoint, [body_params: body_params], options)
+      end
+
+      def delete(endpoint, options \\ []) do
+        Omise.HTTPClient.request(:delete, @key_type, endpoint, [], options)
+      end
+    end
+  end
```
Comment (iporsut) [inline lib/omise/http_client.ex]: Could be able extract `Omise.HTTPClient` module name to variable?

Comment (tteerawat) [reply lib/omise/http_client.ex]: Refactored as suggested here https://github.com/omise/omise-elixir/pull/19/commits/dc8cc812b52359658a62ee4d9877cb8052944bda

File: lib/omise/utils.ex [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+  def normalize_search_params(params) do
+    filters_params = params[:filters] || []
+    params
+    |> Keyword.take([:query])
+    |> Keyword.merge(normalize_filters_params(filters_params))
+  end
+
+  def to_boolean(nil),   do: false
+  def to_boolean(false), do: false
+  def to_boolean(_),     do: true
```
Comment (iporsut) [inline lib/omise/utils.ex]: You can convert any value to bool by `!!`.
```elixir
def to_boolean(val), do: !!val
```

**Comment by tteerawat** [comment]
Not a good idea 🙅‍♂️

**Pull request closed without merging**